Add the Survey step for the improved Jetpack cancellation flow

IanRamosC opened the pull request:
#### Changes proposed in this Pull Request

* This PR adds the component of one of the steps for the improved Jetpack cancellation flow. This is still in progress

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/index.jsx:
```diff
@@ -0,0 +1,265 @@
+		};
+
+		const cancelText = translate( 'Cancel Now' );
+		const cancellingText = translate( 'Cancelling' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Why cancelling?' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/survey-for-jetpack-cancellation-flow)

IanRamosC committed:
Marketing Survey: Add a survey before a Jetpack product is cancelled

This is one of the components that will be shown in the new/improved Jetpack cancellation flow.

IanRamosC committed:
Marketing Survey: Add text input for option 'other'

IanRamosC committed:
Marketing Survey: Fix styles for mobile devices

a8ci18n commented on client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-survey.tsx:
```diff
@@ -0,0 +1,133 @@
+						className="jetpack-cancellation-survey__choice-item-text-input"
+						value={ customAnswerText }
+						onChange={ onChangeCustomAnswerText }
+						placeholder={ translate( 'share your experience' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **27** times:
`translate( 'share your experience…' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/survey-for-jetpack-cancellation-flow)
